auth-jwt-service/src/main/java/io/stargate/auth/jwt/AuthzJwtService.java

```diff
@@ -139,27 +138,23 @@ private JSONObject extractClaimsFromJWT(String token) throws JSONException {
     return payload.getJSONObject(CLAIMS_FIELD);
   }
 
-  private void preCheckDataReadWrite(
-      JSONObject stargateClaims, List<String> primaryKeyValues, Table tableMetadata)
+  private void preCheckDataReadWrite(JSONObject stargateClaims, List<TargetCell> targetCells)
       throws JSONException, UnauthorizedException {
-    List<Column> keys = tableMetadata.primaryKeyColumns();
-
-    if (primaryKeyValues.size() > keys.size()) {
-      throw new IllegalArgumentException("Provided more primary key values than exists");
-    }
-
-    for (int i = 0; i < primaryKeyValues.size(); i++) {
+    for (int i = 0; i < targetCells.size(); i++) {
+      TargetCell targetCell = targetCells.get(0);
       // If one of the columns exist as a field in the JWT claims and the values do not match then
       // the request is not allowed.
-      if (stargateClaims.has(STARGATE_PREFIX + keys.get(i).name())) {
+      if (stargateClaims.has(STARGATE_PREFIX + targetCell.getName())) {
 
-        if (!Column.ofTypeText(keys.get(i).type())) {
+        if (!(targetCell.getType().equals(Type.Varchar.cqlDefinition())
+            || targetCell.getType().equals(Type.Text.cqlDefinition()))) {
```

Comment:
Supporting integers should not be too hard (if it matters), I guess. We only need to convert from the "target" values to `String` for validation.